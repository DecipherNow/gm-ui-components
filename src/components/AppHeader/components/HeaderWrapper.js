import styled from "styled-components";
import { transparentize } from "polished";

import BannerBackgroundImage from "./images/mesh-web.jpg";

const HeaderWrapper = styled.div`
  font-family: ${props => props.theme.FONT_STACK_BASE};
  color: ${props => props.theme.COLOR_CONTENT};
  background-color: ${props => props.theme.COLOR_BACKGROUND_C};
  background-image: linear-gradient(
      to right,
      ${props => props.theme.COLOR_BACKGROUND_C},
      ${props => transparentize(0.7, props.theme.COLOR_BACKGROUND_C)},
      ${props => transparentize(1, props.theme.COLOR_BACKGROUND_C)}
    ),
    url(${BannerBackgroundImage});
  display: flex;
  flex: 0 0 auto;
  flex-direction: column;
  align-items: stretch;
  background-size: 100% 100%, 120% auto;
  background-position: center center, left center;
  background-repeat: no-repeat;
  overflow: hidden;
`;

export default HeaderWrapper;
