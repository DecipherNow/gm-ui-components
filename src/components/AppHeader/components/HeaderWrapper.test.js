import React from "react";
import { shallow } from "enzyme";
import HeaderWrapper from "./HeaderWrapper";

const theme = {
  FONT_STACK_BASE: "sans-serif",
  COLOR_CONTENT: "blue",
  COLOR_BACKGROUND_C: "red"
};

describe("HeaderWrapper", () => {
  it("matches snapshot", () => {
    const aHeaderWrapper = shallow(<HeaderWrapper theme={theme} />);
    expect(aHeaderWrapper).toMatchSnapshot();
  });
});
