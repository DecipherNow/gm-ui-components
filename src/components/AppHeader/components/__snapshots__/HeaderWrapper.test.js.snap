// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderWrapper matches snapshot 1`] = `
.c0 {
  font-family: sans-serif;
  color: blue;
  background-color: red;
  background-image: linear-gradient( to right,red,rgba(255,0,0,0.3),rgba(255,0,0,0) ),url(mesh-web.jpg);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  background-size: 100% 100%,120% auto;
  background-position: center center,left center;
  background-repeat: no-repeat;
  overflow: hidden;
}

<div
  className="c0"
/>
`;
